daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -763,12 +769,148 @@ final class ElementRuntimeData(
+    local, // pathArg: => String,
+    null, //  namespacesArg: => NamespaceBinding,
+    null, // minimizedScopeArg: => NamespaceBinding,
+    null, //defaultBitOrderArg: => BitOrder,
+    None, // optPrimTypeArg: => Option[PrimType],
+    null, // targetNamespaceArg: => NS,
+    NS(namespaceURI), // thisElementsNamespaceArg: => NS,
+    null, // optSimpleTypeRuntimeDataArg: => Option[SimpleTypeRuntimeData],
+    null, // optComplexTypeModelGroupRuntimeDataArg: => Option[ModelGroupRuntimeData],
+    0L, // minOccursArg: => Long,
+    0L, // maxOccursArg: => Long,
+    Nope, // maybeOccursCountKindArg: => Maybe[OccursCountKind],
+    local, // nameArg: => String,
+    null, // targetNamespacePrefixArg: => String,
+    null, // thisElementsNamespacePrefixArg: => String,
+    false, // isHiddenArg: => Boolean,
+    false, // isNillableArg: => Boolean,
+    false, // isArrayArg: => Boolean, // can have more than 1 occurrence
+    false, // isOptionalArg: => Boolean, // can have only 0 or 1 occurrence
+    false, // isRequiredInUnparseInfosetArg: => Boolean, // must have at least 1 occurrence
+    LocalDeclQName(None, local, NS(namespaceURI)), // namedQNameArg: => NamedQName,
+    false, // isRepresentedArg: => Boolean,
+    false, // couldHaveTextArg: => Boolean,
+    0, // alignmentValueInBitsArg: => Int,
+    false, // hasNoSkipRegionsArg: => Boolean,
+    null, // impliedRepresentationArg: => Representation,
+    null, // optIgnoreCaseArg: => Option[YesNo],
+    null, // optDefaultValueArg: => Option[AnyRef],
+    null, // optTruncateSpecifiedLengthStringArg: => Option[Boolean],
+    null, // outputValueCalcExprArg: => Option[CompiledExpression[AnyRef]],
+    Nope, // maybeBinaryFloatRepEvArg: => Maybe[BinaryFloatRepEv],
+    Nope, // maybeByteOrderEvArg: => Maybe[ByteOrderEv],
+    Nope, // maybeFillByteEvArg: => Maybe[FillByteEv],
+    Nope, // maybeCheckByteAndBitOrderEvArg: => Maybe[CheckByteAndBitOrderEv],
+    Nope, // maybeCheckBitOrderAndCharsetEvArg: => Maybe[CheckBitOrderAndCharsetEv],
+    false // isQuasiElementArg: => Boolean
+  ) {
+
+  override def toString() = Misc.getNameFromClass(this) + "(" + this.namedQName.toExtendedSyntax + ")"
+
+  @throws(classOf[java.io.IOException])
+  private def writeObject(out: java.io.ObjectOutputStream): Unit =
+    Assert.usageError("Not for serialization")
+}
+
+/**
+ * Used when unparsing to indicate that a next element event was detected that is
+ * unexpected.
+ */
+final class UnexpectedElementErrorERD(
+  optTRD: Option[TermRuntimeData],
+  local: String,
+  namespaceURI: String,
+  val allPossibleNQNs: Seq[QNameBase])
+  extends ErrorERD(local, namespaceURI) {
+}
+
+/**
+ * Used when unparsing to indicate that multiple elements could be next that
+ * differ only by namespace. This means for some event sources (like JSON) we
+ * don't know which element to create.
+ */
+final class NamespaceAmbiguousElementErrorERD(
+  optTRD: Option[TermRuntimeData],
+  local: String,
```

Comment:
We do actually have a test schema that tests for the warning: combinatiosn_04 in namespaces.tdml. I think the problem though is that there is no way to use a JsonInfosetInputter in the TDML runner to create the infoset. I think we'd need a special unit test that uses JSON. Just pointing out that we at least have a schema that should reproduce this.

Reply:
I t tried to make a little JSON test rig. Where you can unparse from XML to data, parse from data to JSON, then unparse from JSON to data, and compare. 

I wrote it, but I couldn't get it to accept the schemas in the daffodil-tests namespaces tests specifically. I keep getting SDEs having to do with include/import, yet the same schema file imports and includes and works fine running under the TDML runner. 

I will push this as a separate PR.